lib/measured/rails/validations.rb

```diff
@@ -24,7 +24,9 @@ def validate_each(record, attribute, measurable)
 
     if options[:units]
       valid_units = Array(options[:units]).map { |unit| measured_class.unit_system.unit_for(unit) }
-      record.errors.add(attribute, message("is not a valid unit")) unless valid_units.include?(measurable_unit)
+      if valid_units.exclude?(measurable_unit) && record.errors[attribute].exclude?(message("is not a valid unit"))
```

Comment:
Agree with @kmcphillips, that this isn't the best option. You only want to consider `options[:units]` when `measurable_unit` is not `nil`, because when it's `nil` we add this error message on line 23 :)